pkg/controllers/provisioning/scheduling/instance_selection_test.go

```diff
@@ -0,0 +1,453 @@
+			{
+				Key:      v1.LabelOSStable,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"darwin"},
+			},
+		}
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod())
+		node := ExpectScheduled(ctx, env.Client, pod[0])
+		Expect(nodePrice(node)).To(Equal(minPrice))
+		ExpectInstancesWithLabel(cloudProv.CreateCalls[0].InstanceTypes, v1.LabelOSStable, "darwin")
+	})
+	It("should schedule on one of the cheapest instances (pod os = darwin)", func() {
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod(
+			test.PodOptions{NodeRequirements: []v1.NodeSelectorRequirement{{
+				Key:      v1.LabelOSStable,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"darwin"},
+			}}}))
+		node := ExpectScheduled(ctx, env.Client, pod[0])
+		Expect(nodePrice(node)).To(Equal(minPrice))
+		ExpectInstancesWithLabel(cloudProv.CreateCalls[0].InstanceTypes, v1.LabelOSStable, "darwin")
+	})
+	It("should schedule on one of the cheapest instances (pod os = darwin)", func() {
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod(
+			test.PodOptions{NodeRequirements: []v1.NodeSelectorRequirement{{
+				Key:      v1.LabelOSStable,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"darwin"},
+			}}}))
+		node := ExpectScheduled(ctx, env.Client, pod[0])
+		Expect(nodePrice(node)).To(Equal(minPrice))
+		ExpectInstancesWithLabel(cloudProv.CreateCalls[0].InstanceTypes, v1.LabelOSStable, "darwin")
+	})
+	It("should schedule on one of the cheapest instances (prov zone = test-zone-2)", func() {
+		provisioner.Spec.Requirements.Requirements = []v1.NodeSelectorRequirement{
+			{
+				Key:      v1.LabelTopologyZone,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"test-zone-2"},
+			},
+		}
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod())
+		node := ExpectScheduled(ctx, env.Client, pod[0])
+		Expect(nodePrice(node)).To(Equal(minPrice))
+		ExpectInstancesWithLabel(cloudProv.CreateCalls[0].InstanceTypes, v1.LabelTopologyZone, "test-zone-2")
+	})
+	It("should schedule on one of the cheapest instances (pod zone = test-zone-2)", func() {
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner, test.UnschedulablePod(
+			test.PodOptions{NodeRequirements: []v1.NodeSelectorRequirement{{
+				Key:      v1.LabelTopologyZone,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"test-zone-2"},
+			}}}))
+		node := ExpectScheduled(ctx, env.Client, pod[0])
+		Expect(nodePrice(node)).To(Equal(minPrice))
+		ExpectInstancesWithLabel(cloudProv.CreateCalls[0].InstanceTypes, v1.LabelTopologyZone, "test-zone-2")
+	})
+	It("should schedule on one of the cheapest instances (prov ct = spot)", func() {
```

Comment:
Just to follow the above format, should we also test the case (pod ct = spot)?

Reply:
We are but I labeled the test wrong 😆 fixed.